src/HypothesisTests.jl

```diff
@@ -68,35 +68,54 @@ function check_alpha(alpha::Float64)
     end
 end
 
+# Utility to get an optional field (e.g., :tail and :alpha)
+getfield(value::HypothesisTest, name::Symbol, default::Any) =
+    if name in fieldnames(value)
+        Base.getfield(value, name)
+    else
+        default
+    end
+
+get_tail(test::HypothesisTest) = getfield(test, :tail, default_tail(test))
+get_alpha(test::HypothesisTest) = getfield(test, :alpha, 0.05)
+
+# Utility for pretty-printing: Append white space so that length(with_trailing_whitespace(s)) = max(len, length(s))
+with_trailing_whitespace(s::String, len::Int) = s * join(repeat([" "], outer=max(len - length(s), 0)), "")
+
 # Pretty-print
 function Base.show{T<:HypothesisTest}(io::IO, test::T)
     println(io, testname(test))
     println(io, repeat("-", length(testname(test))))
+    
+    alpha = get_alpha(test)
+    tail  = get_tail(test)
+    conf_string = string((1 - alpha) * 100) * "%"
 
     # population details
     has_ci = applicable(StatsBase.confint, test)
+    label_len = has_ci ? length(conf_string) + 21 : 22 # 21 is length of ' confidence interval:', 22 that of 'parameter of interest:'
```

Comment:
There is a function `prettify_detail` inside `Base.show` now that aligns each line consisting of label and value to the longest possible length.